### ISSUE TITLE
Enhancement: member-ordering allows name matching members

### ISSUE THREAD

**Issue by PatrickHallek** [typescript-eslint/typescript-eslint#9630]
### Before You File a Proposal Please Confirm You Have Done The Following...

- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22enhancement%3A+plugin+rule+option%22) and found none that match my proposal.
- [X] I have searched the [current rule list](https://typescript-eslint.io/rules/#rules) and found no rules that match my proposal.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### My proposal is suitable for this project

- [X] I believe my proposal would be useful to the broader TypeScript community (meaning it is not a niche proposal).

### Link to the rule's documentation

https://typescript-eslint.io/rules/member-ordering/

### Description

I propose adding an option to define the member type with a name matching pattern. This enhancement would allow the order of specifically named methods, which might be boilerplate methods of a framework, to be defined.

**Concrete use case:**
I would like to force the `ngOnInit` and `ngOnDestroy` hooks in Angular to be the very top order of methods.

### Fail

```TypeScript
interface Foo {
   someFunction(): void
   ngOnInit(): void
}
```

### Pass

```TypeScript
interface Foo {
   ngOnInit(): void
   someFunction(): void
}
```

**Comment by bradzacher** [comment]
As per the notice at the top of the rule's docs - this rule is considered feature frozen and thus we are not accepting enhancement requests. Thanks for filing.
